src/main/java/uk/ac/cam/cl/dtg/segue/api/UsersFacade.java

```diff
@@ -718,28 +719,41 @@ public Response getUserIdToSchoolMap(@Context final HttpServletRequest httpServl
   @Operation(summary = "Submit a contact form request.")
   public Response requestRoleChange(
       @Context final HttpServletRequest request, final Map<String, String> requestDetails) {
-    if (requestDetails == null || StringUtils.isEmpty(requestDetails.get("verificationDetails"))) {
+    if (requestDetails == null || StringUtils.isEmpty(requestDetails.get("verificationDetails"))
+        || StringUtils.isEmpty(requestDetails.get("userEmail"))) {
       return new SegueErrorResponse(Status.BAD_REQUEST, "Missing form details.").toResponse();
     }
 
+    String userEmail = requestDetails.get("userEmail");
     try {
-      RegisteredUserDTO currentUserDto = this.userManager.getCurrentRegisteredUser(request);
+      RegisteredUser targetUser = userManager.findUserByEmail(userEmail);
+      if (targetUser == null) {
+        // As this endpoint does not require authentication, do not expose whether the requested user exists
+        log.warn("A role change request was made for unknown user: {}", sanitiseExternalLogValue(userEmail));
+        return Response.ok().build();
+      }
 
-      if (currentUserDto.getTeacherPending()) {
+      if (targetUser.getTeacherPending() == Boolean.TRUE) {
```

Comment:
That change was due to a linter warning... `java:S5411`, apparently?
I can revert it if you think that would be better?

Reply:
Just looked up that linter warning and it does make sense, but will <var> == Boolean.TRUE catch the behaviour?

I think you need

`Boolean.TRUE.equals(targetUser.getTeacherPending())`

although it'd be worth checking whether getTeacherPending is actually a boxed bool rather than primitive

https://rules.sonarsource.com/java/type/Code%20Smell/RSPEC-5411/

Reply:
While we've implemented checks to try and ensure that only non-null teacherPending values get committed to the database (and therefore this method should never encounter a null there), the object itself can have a null value there at times (usually when handling registration with a third-party provider). It's not a major imposition here so better safe than sorry?
I'll switch out how the equality is checked though, I'm not sure on that front... 🤔